crypt.go

```diff
@@ -928,9 +926,33 @@ func RenameLUKS2ContainerKey(devicePath, oldName, newName string) error {
 		return errors.New("cannot rename key with unexpected token type")
 	}
 
-	if err := luks2ImportToken(devicePath, newToken, &luks2.ImportTokenOptions{Id: id, Replace: true}); err != nil {
-		return xerrors.Errorf("cannot import new token: %w", err)
+	if replace {
+		if err := luks2ImportToken(devicePath, newToken, &luks2.ImportTokenOptions{Id: id, Replace: true}); err != nil {
+			return xerrors.Errorf("cannot import new token: %w", err)
+		}
+	} else {
+		if err := luks2ImportToken(devicePath, newToken, nil); err != nil {
+			return xerrors.Errorf("cannot import new token: %w", err)
+		}
```

Comment:
@valentindavid what happens to the factory reset process if we reboot here?

Reply:
Probably we should add some tests here. I think what we have here is multiple tokens pointing to the same key slot. So both newName and oldName are pointing to the keyslot.

Reply:
In `internal/luksview/view_test.go`, `TestNewView` actually uses 2 tokens with the same keyslot. Tokens 0 and 1 point to keyslot 0. So I am guessing this is fine.

Reply:
If we end up with a superfluous token because the process gets interrupted, and reprovision is implemented as described in the spec, then a future reprovision with the latest cryptsetup will fix this.